### DISCUSSION TITLE
Command Optional Parameter

### DISCUSSION THREAD

**Discussion by Z3d0X** [nutgram/nutgram#586] [Q&A] [answered]
```php
$bot->onCommand('nickname {nickname}', ...);
```

How can i make a parameter optional?

**Comment by Lukasss93** [comment, marked as answer]
Try this:
```php
$bot->onCommand('nickname {nickname}', ...);
$bot->onCommand('nickname', ...);
```
Or:
```php
$bot->onCommand('nickname ?(.*)?', ...);
```

**Reply by Z3d0X** [reply]
Both works 👍

**Reply by Z3d0X** [reply]
If i can make a suggestion,

Would be awesome if the following syntax is supported for optional parameters "nickname {nickname?}".
It is similar syntax to [laravel's optional route parameters](https://laravel.com/docs/10.x/routing#parameters-optional-parameters)
